subworkflows/nf-core/bam_baserecalibrator/main.nf

```diff
@@ -0,0 +1,59 @@
+//
+// PREPARE RECALIBRATION
+//
+
+include { GATK4_BASERECALIBRATOR  } from '../../../modules/nf-core/gatk4/baserecalibrator/main'
+include { GATK4_GATHERBQSRREPORTS } from '../../../modules/nf-core/gatk4/gatherbqsrreports/main'
+
+workflow BAM_BASERECALIBRATOR {
+    take:
+    ch_cram            // channel: [mandatory] [ meta, cram_markduplicates, crai ]
+    ch_dict            // channel: [mandatory] [ dict ]
+    ch_fasta           // channel: [mandatory] [ fasta ]
+    ch_fasta_fai       // channel: [mandatory] [ fasta_fai ]
+    ch_intervals       // channel: [mandatory] [ intervals, num_intervals ] (or [ [], 0 ] if no intervals)
+    ch_known_sites     // channel: [optional]  [ known_sites ]
+    ch_known_sites_tbi // channel: [optional]  [ known_sites_tbi ]
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Combine cram and intervals for spread and gather strategy
+    ch_cram_intervals = ch_cram.combine(ch_intervals)
+        // Move num_intervals to meta map
+        .map{ meta, cram, crai, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, intervals ] }
```

Comment:
num_intervals was never "approved" . I am all for going for it though and deal with the ramifications later :D

Reply:
Should we leave it as is? I am not sure if I completely get the idea behind it :)